// CIBORG/webapi/router.js
"use strict";

class routerDictionary {
    constructor(handler, description, next) {
      this.handler = handler;
      this.description = description;
      this.next = next;
    }
};

let router = {
    routes: [], // dictionary to registered routes
    root: '/',  // the root URL path of the application
    
    //splt = url.replace(/\s+/, '').split('/');

    // check for duplicated routes
    hasRoute: function(_url) {
        this.routes.forEach(function(route) {
            if(_url.match(route.url)) {
                return true;
            }
        });
        return false;  
    },

    // add command to routes
    add: function(url, handler, description) {
        if(!this.hasRoute(url))
            this.routes.push({ url : url, handler : handler, description : description })
        return this;
    },

    // router navigation from request to function
    get: function(request, response) {},
    post: function(request, response) {},
    put: function(request, response) {},
    delete: function(request, response) {}

}



router.add('GET /games/{name}/groups/{id}', 'services.games.getGamesByGroupID', "GET");
router.add('GET /games/{name}/groups/{id}', 'services.games.getGamesByGroupID', "GET");
router.add('GET /games', 'services.games.getGamesByGroupID', "GET");
console.log(router.routes);
module.exports = router;
