// CIBORG/webapi/router.js
"use strict";
let csgames = require("./../services/games/ciborg-services-games.js");
let csgroups = require("./../services/games/ciborg-services-group.js");

let router = {
    routes: [], // {command, handler, description} it keeps the current registered routes
    root: '/', // the root URL path of the application

    // add command to router, only if handler is of type function
    add: function(command, handler, description) {
        if(typeof command == 'function') {
            handler = command;
            command = '';
        }
        this.routes.push({ command : command, handler : handler, description : description});
    
        return this;
    },

    //  compare the registered entries
    check: function(func) {
        this.routes.forEach(function(route) {
            let match = func.match(route.command);
            if(match) {
                match.shift();
                this.route.handler.apply({}, match);
                return this;
            }  
        });
        return this;
    },

    // navigate to input url
    navigate: function(path) {
        path = path ? path : router.root; // return user to home page
        window.location.href = window.location.href.replace(/#(.*)$/, '') + '#' + path;
        return this;
    }

}

function createRouter() {
    // games
    router.add("GET /games", function(){console.log("GET /games")}, "Get popular games.");
    router.add("GET /games​/{name}", function(){console.log("GET /games​/{name}")}, "Find game by name.");

    // groups
    router.add("POST ​/groups​", function(){console.log("POST ​/groups​")}, "Create group.");
    router.add("PUT /groups​/{id}", function(){console.log("PUT /groups​/{id}")}, "Edit group.");
    router.add("GET ​/groups", function(){console.log("GET ​/groups")}, "List all groups.");
    router.add("GET ​/groups/{id}", function(){console.log("GET ​/groups/{id}")}, "Get group detailed info.");
    router.add("PUT ​/groups​/{id}/games/{name}", function(){console.log("PUT ​/groups​/{id}/games/{name}")}, "Add game to group.");
    router.add("DELETE ​/groups​/{id}/games/{name}", function(){console.log("DELETE ​/groups​/{id}/games/{name}")}, "Remove game from group.");
    router.add("GET ​/groups​/{id}/games", function(){console.log("GET ​/groups​/{id}/games")}, "Get groups games with condition");
}

createRouter();
module.exports = router;
