// CIBORG/webapi/router.js
"use strict";

let router = {
    routes: [], // {command, handler, description} it keeps the current registered routes
    root: '/', // the root URL path of the application

    // add command to router, only if handler is of type function
    add: function(command, handler, description) {
        if(typeof command == 'function') {
            handler = command;
            command = '';
        }
        this.routes.push({ command : command, handler : handler, description : description});
    
        return this;
    },

    //  compare the registered entries
    check: function(func) {
        this.routes.forEach(function(route) {
            let match = func.match(route.command);
            if(match) {
                match.shift();
                this.route.handler.apply({}, match);
                return this;
            }  
        });
        return this;
    },

    // navigate to input url
    navigate: function(path) {
        path = path ? path : router.root; // return user to home page
        window.location.href = window.location.href.replace(/#(.*)$/, '') + '#' + path;
        return this;
    }

}

function createRouter() {
    router.add("GET /games", , "Get popular games.");
    router.add("GET /games​/{name}", , "Find game by name.");
    router.add("GET /games​/{name}​/groups​/{id}", , "Create group.");
    router.add("PUT /games​/{name}​/groups​/{id}", , "Edit group.");
    router.add("GET /games", , "List all groups.");
    router.add("GET /games", , "Get group detailed info.");
    router.add("GET /games", , "Add game to group.");
    router.add("GET /games", , "Remove game from group.");
    router.add("GET /games", , "Get all games from a group, bla bla.");
}

createRouter();
module.exports = router;
